cmd/iam-keys-check/main.go

```diff
@@ -0,0 +1,230 @@
+				_, err = iamClient.GenerateCredentialReport(&iam.GenerateCredentialReportInput{})
+				if err != nil {
+					if aerr, ok = err.(awserr.Error); ok {
+						if aerr.Code() == iam.ErrCodeLimitExceededException {
+							time.Sleep(time.Duration(pollInterval) * time.Millisecond)
+							return getCredentialReport(iamClient, tries-1, pollInterval)
+						}
+					}
+					log.Fatal(err)
+				}
+			} else if aerr.Code() == iam.ErrCodeCredentialReportNotReadyException {
+				time.Sleep(time.Duration(pollInterval) * time.Millisecond)
+				return getCredentialReport(iamClient, tries-1, pollInterval)
+			}
+		}
+		log.Fatal(err)
+	}
+	return report, nil
+}
+
+func sendAlertToSlack(usersSet stringSet, maxDays float64) error {
+
+	payload := map[string]interface{}{
+		"channel": options.SlackChannel,
+		"text":    "AWS notification",
+		"attachments": []map[string]string{
+			map[string]string{
+				"title": "Message",
+				"text":  "The following users have an active access key over " + fmt.Sprint(int(maxDays)) + " days old: " + strings.Join(usersSet.toArray(true), ", ") + "",
+			},
+		},
+	}
+
+	payloadJSON, err := json.Marshal(payload)
+	if err != nil {
+		return err
+	}
+
+	_, err = http.Post(options.SlackWebhookURL, "application/json", bytes.NewBuffer(payloadJSON))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func triggerCheck() {
+
+	maxDays := float64(options.MaxDays)
+
+	if maxDays == 0 {
+		log.Fatal("MaxDays must be greater than 0.")
+	}
+
+	sess := session.MustMakeSession(options.Region, options.Profile)
+	iamClient := iam.New(sess)
+
+	report, err := getCredentialReport(iamClient, 5, int(options.PollInterval))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	reader := csv.NewReader(strings.NewReader(string(report.Content)))
+
+	header, err := reader.Read()
+	if err != nil {
+		if err != io.EOF {
+			log.Fatal(err)
+		}
+	}
+
+	usersSet := newstringSet()
+
+	for {
+
+		inRow, err := reader.Read()
+		if err != nil {
+			if err == io.EOF {
+				break
+			} else {
+				log.Fatal(err)
+			}
+		}
+
+		input := rowToMap(header, inRow)
+
+		if input["access_key_1_active"] == "true" {
+			accessKey1LastRotated, err := time.Parse(timestampFormat, input["access_key_1_last_rotated"])
+			if err != nil {
+				log.Fatal(err)
+			}
+			if days := report.GeneratedTime.Sub(accessKey1LastRotated).Hours() / 24; days > maxDays {
+				usersSet.add(input["user"])
+			}
+		}
+
+		if !usersSet.has(input["user"]) {
+			if input["access_key_2_active"] == "true" {
+				accessKey2LastRotated, err := time.Parse(timestampFormat, input["access_key_2_last_rotated"])
+				if err != nil {
+					log.Fatal(err)
+				}
+				if days := report.GeneratedTime.Sub(accessKey2LastRotated).Hours() / 24; days > maxDays {
+					usersSet.add(input["user"])
+				}
+			}
+		}
+
+	}
+
+	if usersSet.len() > 0 {
+		err = sendAlertToSlack(usersSet, maxDays)
+		if err != nil {
+			log.Fatal(err)
```

Comment:
I like having more context with these Fatal errors. Also relates to my comment above about using zap.

`		logger.Fatal("unable to send alert to slack",
			zap.Error(err))
`

The same applies to the other Fatal calls.

Reply:
done